### ISSUE TITLE
[FEATURE] Support Metadata filtering with Neo4J

### ISSUE THREAD

**Issue by sjivan** [langchain4j/langchain4j#1252]
**Is your feature request related to a problem? Please describe.**
Need the ability to narrow down search within Neo4J graph based on provided metadata 

A good description of the problem and solution with langchain and neo4j is provided here: 

https://medium.com/neo4j/graph-based-metadata-filtering-for-improving-vector-search-in-rag-applications-47fd2efcfc0a

**Comment by jexp** [comment]
Yes sounds like a good addition, we have it in several of the other implementations.
